### ISSUE TITLE
Retrieved a total of '0' driver packages from () matching operational mode: Production

### ISSUE THREAD

**Issue by SAlex007** [SCConfigMgr/ConfigMgr#280]
I installed and configured Driver automation tool, it can download drivers and create sccm packages. AdminService seems to be up and running. I believe I setup the task sequence correctly too but this is what I'm getting in the log when run the TS 

Really appreciate any advice

[ApplyDriverPackage]: Apply Driver Package process initiated	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Apply driver package deployment type: DriverUpdate	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Apply driver package operational mode: Production	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
[PrerequisiteChecker]: Starting environment prerequisite checker	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Supported computer platform detected, script execution allowed to continue	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Supported operating system version currently running detected, script execution allowed to continue	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer manufacturer determined as: Dell	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer model determined as: Latitude 7400	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer SystemSKU determined as: 08E1	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer Fallback SystemSKU determined as: 08E1	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer model detection was successful	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer SystemSKU detection was successful	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Computer details successfully verified	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 - Determined primary computer detection method: SystemSKU	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
[PrerequisiteChecker]: Completed environment prerequisite checker	ApplyDriverPackage	16/10/2020 3:00:15 PM	7360 (0x1CC0)
 **- Retrieved a total of '0' driver packages from () matching operational mode: Production	ApplyDriverPackage	16/10/2020 3:00:16 PM	7360 (0x1CC0)
 - An error occurred while calling  for a list of available driver packages. Error message: The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.	ApplyDriverPackage	16/10/2020 3:00:16 PM	7360 (0x1CC0)
[ApplyDriverPackage]: Apply Driver Package process failed, please refer to previous error or warning messages	ApplyDriverPackage	16/10/2020 3:00:16 PM	7360 (0x1CC0)**

**Comment by jbudd0649** [comment]
@SAlex007  Im seeing the same error, did you ever get this one solved?

**Comment by chaosisbliss** [comment]
Did you assign the Read Only Analyst role to the user you created for the Admin Service?

**Comment by jbudd0649** [comment]
I did indeed.

**Comment by chaosisbliss** [comment]
What happens when you take the. Invoke-CMApplyDriverPackage.ps1 and manually run commands in debugging mode?  
The commands you need are at the bottom. You can test this from any machine. 
https://msendpointmgr.com/modern-driver-management/

**Comment by jbudd0649** [comment]
When I run the debug command nothing happens. I checked c:\Windows\temp and I don't see a driver package.
![mddeploy](https://user-images.githubusercontent.com/39967141/98413745-21b24f00-2048-11eb-9980-8de9ddbc6c73.jpg)

**Comment by chaosisbliss** [comment]
It's in your screenshot. Open the ApplyDriverPackage.log in the temp folder.

**Comment by jbudd0649** [comment]
Wow. I completely overlooked that. Here's the log. 
[ApplyDriverPackage.log](https://github.com/MSEndpointMgr/ConfigMgr/files/5503487/ApplyDriverPackage.log)

**Comment by jbudd0649** [comment]
![packages](https://user-images.githubusercontent.com/39967141/98414405-4ce96e00-2049-11eb-908c-24e32c87891b.jpg)

**Comment by chaosisbliss** [comment]
This really looks like account permissions.  Can you verify the AD user you created has the correct roles etc in ConfigMGR?

**Comment by jbudd0649** [comment]
I used my domain admin account and my new service account. Both are SCCM admins.
![accounts](https://user-images.githubusercontent.com/39967141/98415576-86bb7400-204b-11eb-9451-ccfa12b5cea1.jpg)

**Comment by chaosisbliss** [comment]
What are the switches you are using in the task sequence? Be sure to coverup your password and server name when you provide them.

**Comment by jbudd0649** [comment]
![mdm task sequence](https://user-images.githubusercontent.com/39967141/98543419-7af4cb00-2260-11eb-93e3-3a69fcb3b5e4.jpg)
![mdm variables](https://user-images.githubusercontent.com/39967141/98543426-7d572500-2260-11eb-98d8-132d11b124fd.jpg)
![apply drivers](https://user-images.githubusercontent.com/39967141/98543434-7fb97f00-2260-11eb-8f8f-aab2ff5d516a.jpg)

**Comment by chaosisbliss** [comment]
That looks good. Do you have two sites or something? I noticed that your packages refer to two different site codes. Have you setup permissions for both sites?

**Comment by jbudd0649** [comment]
I only have 1 site, We got a new server a year or two ago. When I set it up I did a site migration from the old site to the new one. P01 is the current site. All of the packages that I migrated came over with the original site code.

**Comment by chaosisbliss** [comment]
Ok, that makes sense. Are all of these packages were created with the new driver autmation tool 6.4.9? You are on what version of ConfigMgr?

**Comment by jbudd0649** [comment]
The DAT tool says version 6.4.9

Configmanager is on version 2006

**Comment by jbudd0649** [comment]
I checked to make sure The SMS_REST_Provider service is running and I am able to browse to the url in chrome.

**Comment by chaosisbliss** [comment]
Can you browse to these without issues? If you can browse to the second link, perform a manual search for one of the devices failing in your applydriverpackage.log file. 

https://server.fqdn/AdminService/wmi/
https://server.fqdn/AdminService/wmi/SMS_Package?$filter=contains(Name,%27Drivers%27)

**Comment by jbudd0649** [comment]
Yes, I can get to both URL's and when I do a page find for "Optiplex 7010" I am able to locate several references and packages. I would send a screencap, but, my server name appears a lot on this page.

**Comment by jbudd0649** [comment]
Here is a small slice


![admin service](https://user-images.githubusercontent.com/39967141/98560646-fa40c980-2275-11eb-9020-d99e8519a2cf.jpg)

**Comment by chaosisbliss** [comment]
Nah. You don't have to send a screenshot.  I believe you.  I had this exact same problem though.  It ended up being I assigned the wrong role to the user.  I had a read-only Analyst and a read-only Reporting analyst role I manually setup.  I incorrectly assigned the one I manually created lol.

**Comment by jbudd0649** [comment]
Here is the package referenced above. IT is distributed to all my DP's.  I browsed to the source folder as well to make sure the drivers were present and they are.

![package id](https://user-images.githubusercontent.com/39967141/98560932-4be95400-2276-11eb-8523-4306adb9a48d.jpg)

**Comment by chaosisbliss** [comment]
I am honestly out of ideas. Maybe someone smarter could join in lol.

**Comment by jbudd0649** [comment]
Thanks for the help, at least I'm not the only one stumped.

**Comment by chaosisbliss** [comment]
I know this sounds stupid, but could you assign the Read-Only Analyst role to your MDMUser? Then, just run that .\invoke with the debug commands.

**Comment by jbudd0649** [comment]
No difference.

**Comment by jbudd0649** [comment]
I think I may have fixed this. Check out this thread
https://github.com/MSEndpointMgr/ModernDriverManagement/issues/35

Looks like I have an old version of the script, which is weird because I got it from https://msendpointmgr.com/modern-driver-management/

I was able to use the script in debug mode. Im going to try an OSD and see what happens.

**Comment by jbudd0649** [comment]
Success. Thanks for all the help @chaosisbliss

**Comment by chaosisbliss** [comment]
Fantasitc! Hope you at least learned something from everything I threw at you lol.
